WordPress/Classes/ViewRelated/Site Creation/Final Assembly/SiteAssemblyWizardContent.swift

```diff
@@ -185,30 +190,63 @@ final class SiteAssemblyWizardContent: UIViewController {
     private func installErrorStateViewController(with type: ErrorStateViewType) {
         var configuration = ErrorStateViewConfiguration.configuration(type: type)
 
-        configuration.contactSupportActionHandler = { [weak self] in
+        configuration.retryActionHandler = { [weak self] in
             guard let self = self else {
                 return
             }
-            self.contactSupportTapped()
+            self.retryTapped()
         }
 
+        self.installErrorStateViewController(with: configuration)
+    }
+
+    private func installDomainCheckoutErrorStateViewController(domain: DomainSuggestion, site: Blog) {
+        var configuration = ErrorStateViewConfiguration.configuration(type: .domainCheckoutFailed)
+
         configuration.retryActionHandler = { [weak self] in
-            guard let self = self else {
+            guard let self else {
                 return
             }
-            self.retryTapped()
+            self.contentView.status = .inProgress
+            self.attemptDomainPurchasing(domain: domain, site: site)
         }
 
-        configuration.dismissalActionHandler = { [weak self] in
-            guard let self = self else {
-                return
+        self.installErrorStateViewController(with: configuration)
+    }
+
+    private func installErrorStateViewController(with configuration: ErrorStateViewConfiguration) {
+        var configuration = configuration
+
+        if configuration.contactSupportActionHandler == nil {
+            configuration.contactSupportActionHandler = { [weak self] in
+                guard let self = self else {
+                    return
+                }
+                self.contactSupportTapped()
             }
-            self.dismissTapped()
         }
 
+        if configuration.dismissalActionHandler == nil {
+            configuration.dismissalActionHandler = { [weak self] in
+                guard let self = self else {
+                    return
+                }
+                self.dismissTapped()
+            }
+        }
```

Comment:
The code above sets default `dismissal` and `contact support` action handlers if those closures are `nil`.

Reply:
Thanks for the explanation.